crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+            ))
+        })?;
+
+        // Extract the inputs.
+        let input_start = extract_relocatable(vm, input_start)?;
+        let input_end = extract_relocatable(vm, input_end)?;
+        let inputs = vm_get_range(vm, input_start, input_end).map_err(|_| {
+            HintError::CustomHint(Box::from("Failed to read input data".to_string()))
+        })?;
+
+        self.match_cheatcode_by_selector(vm, selector, &inputs, output_start, output_end)
+            .map_err(Into::into)
+    }
+
+    #[allow(
+        unused,
+        clippy::too_many_lines,
+        clippy::trivially_copy_pass_by_ref,
+        clippy::too_many_arguments
+    )]
+    fn match_cheatcode_by_selector(
+        &mut self,
+        vm: &mut VirtualMachine,
+        selector: &str,
+        inputs: &[Felt252],
+        output_start: &CellRef,
+        output_end: &CellRef,
+    ) -> Result<(), EnhancedHintError> {
+        let mut buffer = MemBuffer::new_segment(vm);
+        let result_start = buffer.ptr;
+
+        match selector {
+            "call" => {
+                let contract_address = inputs[0].to_field_element();
+                let function_name = as_cairo_short_string(&inputs[1])
+                    .expect("Failed to convert function name to short string");
+                let calldata_length = inputs[2]
+                    .to_usize()
+                    .expect("Failed to convert calldata length to usize");
+                let calldata = Vec::from(&inputs[3..(3 + calldata_length)]);
+                let calldata_felts: Vec<FieldElement> = calldata
+                    .iter()
+                    .map(StarknetConversions::to_field_element)
+                    .collect();
+
+                let call_response = self.runtime.block_on(call::call(
+                    contract_address,
+                    &function_name,
+                    calldata_felts,
+                    self.provider,
+                    &BlockId::Tag(Pending),
+                ))?;
+
+                buffer
+                    .write(call_response.data.len())
+                    .expect("Failed to insert data length");
+
+                buffer
+                    .write_data(
+                        call_response
+                            .data
+                            .iter()
+                            .map(StarknetConversions::to_felt252),
+                    )
+                    .expect("Failed to insert data");
+
+                Ok(())
+            }
+            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+        }?;
+
+        let result_end = buffer.ptr;
+        insert_value_to_cellref!(vm, output_start, result_start)?;
+        insert_value_to_cellref!(vm, output_end, result_end)?;
+
+        Ok(())
+    }
+}
+
+pub fn run(
+    script_path: &Utf8PathBuf,
+    provider: &JsonRpcClient<HttpTransport>,
+    runtime: Runtime,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
```

Comment:
not sure if we should allow that, I think we should just take the script path and try to search for its scarb toml based on it. Allowing to pass this could potentially lead to some confusion and maybe even some errors. wdyt?

Reply:
good point!

Reply:
done